x-pack/plugins/fleet/server/services/agents/crud.ts

```diff
@@ -512,6 +512,11 @@ export async function getAgentVersionsForAgentPolicyIds(
                   policy_id: agentPolicyIds,
                 },
               },
+              {
+                term: {
+                  active: true,
```

Comment:
didn't we mean to exclude managed offline fleet-servers as a special case to enable secrets where the deployment was upgraded in cloud?

Reply:
I did go down that path initially, but it seemed like we might have to rewrite this query in order to accomplish it. Realizing now after looking with a fresh set of eyes that there might be a better way by removing the `size: 0` and looking at the actual policy query results in this query. Let me get a commit together and see how that goes.

Reply:
Slightly challenging part of this is that the `is_managed` property doesn't exist in the `.fleet-agents` index at all, so we need to re-query all policies to check whether they're managed. This could cause performance issues in an environment with many policies, I think - but I will see what can be done. 

Also, should we apply this logic for all managed policies, or only the `policy-elastic-agent-on-cloud` policy? I don't think we have anything that exists today where we apply special logic based on the cloud policy ID, so I wonder if the managed check is safer.

Reply:
I tend to agree with the general `is_managed` check, since there might be managed policies used in other environments like ECK. Though I'm not sure how much managed policies are being used outside of Elastic Cloud.